scripts/package/fpsci_packager_generator.py

```diff
@@ -0,0 +1,125 @@
+# This script generates a packaging script based on the files used by the last successful run of the program.
+#
+# This script should be run from the root directory of a checkout out FPSci repository. 
+# Normally this means that you'd envoke it with `python scripts/package/fpsci_packager_generator.py`.
+# This script outputs a new version of `scripts/package/fpsci_packager.sh` based on the latest `data-files/log.txt`.
+# Both of those can be changed in the parameters below
+#
+# Once generated, `scripts/package/fpsci_packager.sh` should be run using bash from the root of the repository.
+# It will create a `dist/` directory and copy all of the needed files into it.
+#
+# The intention is that the contents of the `dist/` directory can then be zipped up for distribution.
+
+# Current release being built. Defaults to the git hash. Use "master", "branch name" or "tag name" for other builds.
+import subprocess
+currentRelease = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).strip().decode()
+# currentRelease = "master"
+
+# Name of the packaging script to generate
+outputScript = 'scripts/package/fpsci_packager.sh'
+
+# Path of the log.txt file to use
+inputLog = 'data-files/log.txt'
+
+# The list of files that should always be excluded
+excludeList = ['log.txt', 'keymap.Any', 'systemconfig.Any']
+# The list of files that should be excluded for release builds, but may want to be included for specific experiment distributions
+# Use the empty list if you want to distribute an experiment
+secondaryExcludeList = ['experimentconfig.Any', 'startupconfig.Any', 'userconfig.Any', 'userstatus.Any']
+# secondaryExcludeList = []
+
+# List of paths to search for to merge into the same distribution directory
+basePath = ['data-files', 'game', 'common']
+
+# Distribution path - where the output script will generate the distribution
+distPath = 'dist/'
+
+# Automatic configuation
+import os
+g3dPath = os.environ.get('g3d').replace('\\','/')
+runPath = os.getcwd().replace('\\','/')
+
+########## End script configuration ############
+
+from datetime import datetime
+now = datetime.now()
+
+if __name__ == '__main__':
+    # The log file which lists the files used
+    log = open(inputLog)
+
+    # Get the current set of lines in the script
+    lineSet = set(open(outputScript))
+    for line in sorted(lineSet):
+        if line.startswith('#'):
+            lineSet.remove(line)
+
+    # Hard coded files to copy
+    lineSet.add('mkdir -p ' + distPath + '\n')
+    lineSet.add('cp data-files/g3d-license.txt '  + distPath + '\n')
+    lineSet.add('cp LICENSE.txt '  + distPath + '\n')
+    lineSet.add('cp README.txt '  + distPath + '\n')
```

Comment:
I prefer to keep these here since these needed to be manually added to get this to work from a fresh `log.txt` and I'd rather have no-op adds than forget to add them again at some point in the future.